ext/TimeVaryingInputs0DExt.jl

```diff
@@ -227,4 +241,53 @@ function TimeVaryingInputs.evaluate!(
     )
 end
 
+function TimeVaryingInputs.evaluate!(
+    destination,
+    itp::InterpolatingTimeVaryingInput0D{<:AbstractArray{<:ITime}},
+    date::DateTime,
+    args...;
+    kwargs...,
+)
+    t0_date = DateTime(itp.range[1])
+    diff_ms = date - t0_date
+    converted_time =
+        ITime(diff_ms.value; period = typeof(diff_ms)(1), epoch = t0_date)
+    return TimeVaryingInputs.evaluate!(
+        destination,
+        itp,
+        converted_time,
+        args...;
+        kwargs...,
+    )
+
+end
+
+function TimeVaryingInputs.evaluate!(
+    destination,
+    itp::InterpolatingTimeVaryingInput0D{<:AbstractArray{<:ITime}},
+    time::Number,
+    args...;
+    kwargs...,
+)
+    converted_time = promote(ITime(time), itp.range[1])[1]
+    return TimeVaryingInputs.evaluate!(
```

Comment:
This is just style, but I think
```suggestion
    converted_time = first(promote(ITime(time), itp.range[1]))
    return TimeVaryingInputs.evaluate!(
```
is a little cleaner.